src/controller/CommissioningDelegate.h

```diff
@@ -77,38 +77,129 @@ class CommissioningParameters
     static constexpr size_t kMaxSsidLen          = 32;
     static constexpr size_t kMaxCredentialsLen   = 64;
 
+    // Value to use when setting the commissioning failsafe timer on the node being commissioned.
+    // If the failsafe timer value is passed in as part of the commissioning parameters, that value will be used. If not supplied,
+    // the AutoCommissioner will set this to the recommended value read from the node. If that is not set, it will fall back to the
```

Comment:
This is a critique that falls outside the scope of this PR.  However, I find it odd that we have a spec feature where nodes would have a recommended value for the fail-safe timer.  How on earth will they know what this should be?  Sure, they can have a notion about e.g. how fast they can do a CASE handshake.

But the commissioner will likely be calling off-node to the DCL and into signing infrastructure.  Commissionees can't account for how long this will take.  Why would they even try?

Reply:
It's an interesting point, especially because the commissioning process was written before the device attestation discussion was finished. That being said, the call out to the DCL doesn't need to happen while the failsafe is armed. The spec just specifies that the commissioners trust store should include all the PAAs in the DCL, not that it needs to call out and check that during commissioning. I expect that if the commissioner is calling the outside world while the failsafe is armed, it should add some time to the timer because the device would not account for that. But right now we're pulling from an internal trust store.

That being said, the only parts of the commissioning process that REALLY need to be protected by the failsafe are the parts that make changes on the device (configuring, installing certs, adding networks). Device attestation doesn't make device changes, it just reads the certs and asks for a signature. That could come before failsafe is armed IF we can guarantee that devices doesn't require the regulatory information to supply the certs. But that's definitely a spec change and would need to go through that process.

Reply:
Opened an issue here to track https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/5037. Not sure we'll get to this by the FC cutoff, but still good to have a record here.